engine/access/state_stream/backend/backend_account_statuses.go

```diff
@@ -80,38 +87,69 @@ func (b *AccountStatusesBackend) SubscribeAccountStatusesFromStartHeight(
 func (b *AccountStatusesBackend) SubscribeAccountStatusesFromLatestBlock(
 	ctx context.Context,
 	filter state_stream.AccountStatusFilter,
+	criteria optimistic_sync.Criteria,
 ) subscription.Subscription {
 	nextHeight, err := b.executionDataTracker.GetStartHeightFromLatest(ctx)
 	if err != nil {
 		return subscription.NewFailedSubscription(err, "could not get start height from latest")
 	}
-	return b.subscribe(ctx, nextHeight, filter)
+	return b.subscribe(ctx, nextHeight, filter, criteria)
 }
 
-// getAccountStatusResponseFactory returns a function that returns the account statuses response for a given height.
+// getAccountStatusResponseFactory returns a closure that produces account status responses for
+// a given block height while preserving fork affinity across calls in a single stream.
+//
+// The returned function is designed to be invoked sequentially by the streaming layer
+// (e.g., `subscription.Next`). It threads the execution lineage by carrying forward the `ExecutionResultID`
+// from the previous successful call and placing it into the `optimistic_sync.Criteria` as
+// `ParentExecutionResultID` for the next call. This makes subsequent lookups prefer data
+// from the same execution fork, ensuring consistency during forks.
+//
+// The returned closure captures an internal `nextCriteria` value updated after each successful call.
```

Comment:
Adjusted wording, and removed some implementation details that the caller doesn't need to know.
```suggestion
// (e.g., `subscription.Next`). It guarantees that only data from a single execution fork is
// returned by ensuring that the `PreviousResultID` from each successive execution result
// matches the ID of the execution result used to produce the last message returned on the
// stream.
```